vars/neoDeploy.groovy

```diff
@@ -150,6 +155,17 @@ void call(parameters = [:]) {
             // since the map did not change, it is not required to replace the previous configuration map.
             .use()
 
+        Set extensionFileNames
+
+        if(configuration.extensions == null) {
```

Comment:
I think this is important.

> Use the safe dereference operator ?., to simplify the code for accessing objects and object members safely. Using this operator, the Groovy compiler checks null objects and null object members, and returns null if the object or the object member is null and never throws a NullPointerException.

Reply:
Configuration is never null. In case configuration is `null` this is an real issue. According to the fail fast approach it is OK to fail here with NPE rather than continuing with an undefind/inappropriate state.

I think using the `?.` operator is appropriate in case where it is a valid state that a variable is null. From that point of view it can be considered as a mistake to use the `?.` in situations where a variable is expected to hold a value.

Reply:
Yes, that's true. Then, I would simply use the Groovy truth `configuration.extensions`, but if you don't want to use Groovy syntax ...

Reply:
Also I would initialize `extensionFileNames`. It is null, or I would initialize it in default_pipeline_environment.yml.

Reply:
Also I would initialize `configuration.extensions` in default_pipeline_environment.yml.

Reply:
And if you can just have an if, why an if-else?

```
Set extensionList = []

if(configuration.extensions) {
    extensionFileNames = configuration.extensionsin Collection ? configuration.extensions: [configuration.extensions]
}
```